sui-execution/latest/sui-adapter/src/static_programmable_transactions/metering/translation_meter.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::gas_charger::GasCharger;
+use sui_protocol_config::ProtocolConfig;
+use sui_types::error::{ExecutionError, ExecutionErrorKind};
+
+/// The [`TranslationMeter`] is responsible for metering gas usage for various operations
+/// during the translation of a transaction. It interacts with and exposes interfaces to the
+/// [`GasCharger`] it holds in order to deduct gas based on the operations performed.
+///
+/// It holds a reference to the `ProtocolConfig` to access protocol-specific configuration
+/// parameters that may influence gas costs and limits.
+pub struct TranslationMeter<'pc, 'gas> {
+    protocol_config: &'pc ProtocolConfig,
+    charger: &'gas mut GasCharger,
+    charged: u64,
+}
+
+impl<'pc, 'gas> TranslationMeter<'pc, 'gas> {
+    pub fn new(
+        protocol_config: &'pc ProtocolConfig,
+        gas_charger: &'gas mut GasCharger,
+    ) -> TranslationMeter<'pc, 'gas> {
+        TranslationMeter {
+            protocol_config,
+            charger: gas_charger,
+            charged: 0,
+        }
+    }
+
+    pub fn charge_base_inputs(&mut self, num_inputs: usize) -> Result<(), ExecutionError> {
+        let amount = (num_inputs as u64)
+            .max(1)
+            .saturating_mul(self.protocol_config.translation_per_input_base_charge());
+        debug_assert!(amount > 0);
+        self.charger
+            .move_gas_status_mut()
+            .deduct_gas(amount.into())
+            .map_err(Self::gas_error)
```

Comment:
Should this be `self.charge`?

Reply:
yup yup yup. It should indeed. Good catch!